src/components/FormSubmit/index.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import lodashGet from 'lodash/get';
+import {View} from 'react-native';
+import {propTypes, defaultProps} from './formSubmitPropTypes';
+
+class FormSubmit extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.submitForm = this.submitForm.bind(this);
+    }
+
+    submitForm(event) {
+        if (event.shiftKey || event.key !== 'Enter') {
+            return;
+        }
+
+        const tagName = lodashGet(event, 'target.tagName', '');
+
+        if (tagName === 'INPUT' || tagName === 'SELECT') {
+            this.props.onSubmit();
+            return;
+        }
+
+        if (tagName === 'TEXTAREA' && lodashGet(event, 'target.dataset.blurOnSubmit', 'false') === 'true') {
+            this.props.onSubmit();
+            return;
+        }
+
+        if (lodashGet(event, 'target.role') === 'checkbox') {
+            this.props.onSubmit();
+        }
+    }
+
+    render() {
+        return (
+            <View onKeyDownCapture={this.submitForm} style={this.props.style}>{this.props.children}</View>
```

Comment:
comment on why we have to listen for capture phase here

Reply:
Can you help me with comment. Thanks!

Reply:
OK, I think you can paste this as a comment. 
```
react-native-web prevents event bubbling on TextInput for key presses [necolas/react-native-web@fa47f80/packages/react-native-web/src/exports/TextInput/index.js#L272](https://github.com/necolas/react-native-web/blob/fa47f80d34ee6cde2536b2a2241e326f84b633c4/packages/react-native-web/src/exports/TextInput/index.js#L272). This use capture phase.